core/crypto/crypto.h

```diff
@@ -87,6 +104,11 @@ class Crypto : public Reference {
 	virtual bool verify(HashingContext::HashType p_hash_type, Vector<uint8_t> p_hash, Vector<uint8_t> p_signature, Ref<CryptoKey> p_key) = 0;
 	virtual Vector<uint8_t> encrypt(Ref<CryptoKey> p_key, Vector<uint8_t> p_plaintext) = 0;
 	virtual Vector<uint8_t> decrypt(Ref<CryptoKey> p_key, Vector<uint8_t> p_ciphertext) = 0;
+	virtual PackedByteArray hmac_digest(HashingContext::HashType p_hash_type, PackedByteArray p_key, PackedByteArray p_msg) = 0;
+
+	// Compares two HMACS for equality without leaking timing information in order to prevent timing attakcs.
+	// @see: https://paragonie.com/blog/2015/11/preventing-timing-attacks-on-string-comparison-with-double-hmac-strategy
+	bool hmac_equals(PackedByteArray p_hm1, PackedByteArray p_hm2);
```

Comment:
Would it make sense to call this function: `constant_time_compare`? `safe_bytes_compare`? MbedTLS itslef has an internal `mbedtls_ssl_safer_memcmp` that they plan to expose in future versions: https://github.com/ARMmbed/mbedtls/issues/3649 (and which we could use in the future).

Reply:
I'm fine with this. I'm not super picky about the name. I'll rename to `constant_time_compare`.